### PR TITLE
Disable automatic scalafmt

### PR DESCRIPTION
**Pull request by eatkins**
This is unbearably slow.

- [x] I've read the [CONTRIBUTING](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) guidelines

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -39,7 +39,7 @@ def buildLevelSettings: Seq[Setting[_]] =
       homepage := Some(url("https://github.com/sbt/sbt")),
       scmInfo := Some(ScmInfo(url("https://github.com/sbt/sbt"), "git@github.com:sbt/sbt.git")),
       resolvers += Resolver.mavenLocal,
-      scalafmtOnCompile := true,
+      scalafmtOnCompile := false,
```
Comment (eed3si9n) [inline build.sbt:42]: Could you update CONTRIBUTING and maybe even the `onLoadMessage` to tell ppl to run mima check and Scalafmt commands before committing?

**Review by eed3si9n — CHANGES_REQUESTED** [review]

**Comment by eatkins** [comment]
Oh, yes. That makes sense, we should also add the scalafmt check to travis as @xuwei-k suggested.

**Comment by dwijnand** [comment]
You could give sbt-scalafmt 2.0.0 a try.  It includes performance improvements: https://github.com/scalameta/sbt-scalafmt/releases/tag/v2.0.0, both in using zinc's analysis to make scalafmt incremental and caching of the Scalafmt object.

**Comment by eed3si9n** [comment]
closing in favor of https://github.com/sbt/sbt/pull/4623

**Pull request closed without merging**